client/src/main/java/io/micrometer/prometheus/rsocket/PrometheusRSocketClient.java

```diff
@@ -150,15 +156,57 @@ public void close() {
     }
   }
 
+  /**
+   * Pushes the data in a blocking way and closes the connection.
+   */
+  public void pushAndCloseBlockingly() {
+    pushAndCloseBlockingly(Duration.ofSeconds(5));
+  }
+
+  /**
+   * Pushes the data in a blocking way and closes the connection.
+   * @param timeout the amount of time to wait for the data to be sent
+   */
+  public void pushAndCloseBlockingly(Duration timeout) {
+    CountDownLatch latch = new CountDownLatch(1);
+    PublicKey key = latestKey.get();
+    if (key != null) {
+      try {
+        sendingSocket
+            .requestResponse(scrapePayload(key))
```

Comment:
This should be `requestResponse` instead of `fireAndForget` since the latter will almost immediately emit the complete signal (before anything happens on the wire) so waiting for it to terminate does not make too much sense; `requestResponse` on the other hand will wait for the response from the server.

Also, since this should be called only on shutdown in a separate thread (shutdown hook):
- This should not have too much of an impact on non-ephemeral (long running) apps (the worst case is delaying the shutdown with a configurable timeout but users can also move to the non-blocking version)
- Since since ephemeral apps shutdown a lot the impact is higher on them (still should be low/minimal) but in their case, this feature is more important so this tradeoff seems reasonable (timeout is configurable and users can move to the non-blocking version)

Reply:
Could we leave a comment or reference in the code so someone doesn't come along later and think "hey, this could be fire-and-forget!"?